segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentMetadataTests.java

```diff
@@ -212,6 +218,47 @@ private void checkAttributesEqual(Map<UUID, Long> expected, Map<UUID, Long> actu
         }
     }
 
+    /**
+     * Validates that we can safely iterate over a StreamSegmentMetadata.AttributesView elements while the backing
+     * StreamSegmentMetadata maps are being modified.
+     */
+    @Test
+    public void testAttributeConcurrentOperations() throws ExecutionException, InterruptedException, TimeoutException {
+        StreamSegmentMetadata metadata = new StreamSegmentMetadata(SEGMENT_NAME, SEGMENT_ID, CONTAINER_ID);
+
+        // Step 1: initial set of attributes.
+        Random rnd = new Random(0);
+        val expectedAttributes = generateAttributes(rnd);
+
+        metadata.updateAttributes(expectedAttributes);
+        ExecutorService executorService = null;
+        CompletableFuture<Void> iterateOverView;
+        CompletableFuture<Void> modifyBackingAttributes;
+        try {
+            executorService = ExecutorServiceHelpers.newScheduledThreadPool(4, "testAttributes");
```

Comment:
You can just do `@Cleanup("shutdownNow")` to avoid the `try-finally`. We do that all over the place.

Reply:
This has been removed in the new test.